test/infrastructure/inmemory/pkg/server/api/watch.go

```diff
@@ -149,6 +205,18 @@ func (m *WatchEventDispatcher) Run(ctx context.Context, timeout string, w http.R
 	ctx, cancel := context.WithTimeout(ctx, seconds)
 	defer cancel()
 	defer timeoutTimer.Stop()
+
+	// Determine the highest written resourceVersion so we can filter out duplicated events from the channel.
+	minResourceVersion := uint64(0)
+	if len(initialEvents) > 0 {
+		minResourceVersion, err = strconv.ParseUint(initialEvents[len(initialEvents)-1].Object.GetResourceVersion(), 10, 64)
```

Comment:
Instead of assuming an ordering of events by resource version (which I'm not sure we are enforcing somewhere), what about computing minResourceVersion when we go through initialEvents in the for loop above

Reply:
we can now simply use the resourceVersion from the v1List :-)

Reply:
Note: updated to that.

Also great catch!